model/src/main/kotlin/Project.kt

```diff
@@ -68,6 +68,10 @@ data class Project(
          */
         val scopes: SortedSet<Scope>
 ) : Comparable<Project> {
+    fun collectAllDependencies(): SortedSet<Identifier> = sortedSetOf<Identifier>().also { result ->
```

Comment:
Here the return type can be omitted as it's not recursive like in `PackageReference`.

Reply:
But then Kotlin infers the type TreeSet<String> which I didn't like.